bin/libs/component.sh

```diff
@@ -408,6 +408,154 @@ process_zss_plugin_install() {
+}
+
+# $1 = key=value
+get_value_of_string() {
+  echo "$1" | sed 's/=/ /g' | awk '{print $2}'
+}
+
+# $1 = line number
+# $2 = file
+get_string_at_line_number() {
+  awk -v n="$1" 'NR == n {print $0}' "$2"
+}
+
+# $1 = search_key
+# $2 = file
+get_line_number_of_key() {
+  grep "$1" "$2" > /dev/null
+  if [ $? -eq 0 ]; then
+    awk -v s="$1" '$0 ~ s {print NR}' "$2"
+  else
+    echo ""
+  fi
+}
+
+# $1 = line number
+# $2 = file
+remove_key_value_at_line_number() {
+  mv "$2" "$2.tmp"
+  awk -v n=$1 'NR != n' "$2.tmp" > "$2"
+  rm -f "$2.tmp"
+}
+
+# $1 = key=value
+# $2 = file
+add_key_value_at_end_of_file() {
+  echo "$1" >> "$2"
+}
+
+zis_plugin_install() {
+  plugin_path="${1}"
+  zwes_zis_pluginlib="${2}"
+  zwes_zis_parmlib="${3}"
+  zwes_zis_parmlib_member="${4}"
+  plugin_id="${5}"
+  component_dir="${6}"
+  parmlib_member_as_unix_file="./${zwes_zis_parmlib_member}.txt"
+
+  if [ ! -f "$parmlib_member_as_unix_file" ]; then
+    copy_mvs_to_uss "${zwes_zis_parmlib}(${zwes_zis_parmlib_member})" "${parmlib_member_as_unix_file}"
+  fi
+
+  changed=0
+
+  if [ -d "${component_dir}${plugin_path}" ]; then
+    if [ -d "${component_dir}${plugin_path}/loadlib" ] && [ -d "${component_dir}${plugin_path}/samplib" ]; then
+      for module in $(ls ${component_dir}${plugin_path}/loadlib); do
+        copy_uss_to_mvs "${component_dir}${plugin_path}/loadlib/${module}" "$zwes_zis_pluginlib"
+        if [ $? != 0 ]; then
+          exit 1
+        fi
+      done
+      for params in $(ls ${component_dir}${plugin_path}/samplib); do
+        if [ ! -f "${component_dir}${plugin_path}/samplib/${params}" ]; then
+          exit 2
+        fi
+        while read samplib_key_value; do
+          prefix=$(echo "$samplib_key_value" | cut -c -2)
+          if [ "$prefix" = "//" ] || [ "$prefix" = "* " ] || [ "$prefix" = "" ]; then
+            continue
+          fi
+          grep -x "$samplib_key_value" "$parmlib_member_as_unix_file" > /dev/null
+          if [ $? -eq 0 ]; then
+            echo "The key-value pair $samplib_key_value is being skipped because it's already there and hasn't changed."
+            continue
+          else
+            samplib_key=$(get_key_of_string "$samplib_key_value")
+            if [ "$samplib_key" = "" ]; then
+              exit 3
+            fi
+            # In the case of a key not being there, an empty string will be returned.
+            num=$(get_line_number_of_key "$samplib_key" "$parmlib_member_as_unix_file")
+            if [ "$num" != "" ]; then
+              remove_key_value_at_line_number "$num" "$parmlib_member_as_unix_file"
+              add_key_value_at_end_of_file "$samplib_key_value" "$parmlib_member_as_unix_file"
+              changed=1
+            else
+              # The key doesn't exist. Just add the key-value pair to the end of the file.
+              add_key_value_at_end_of_file "$samplib_key_value" "$parmlib_member_as_unix_file"
+              changed=1
+            fi
+          fi
```

Comment:
Think this whole section can be wrapped into a function `update_uss_parmlib_key(parmlib_file, key, value)`.